frontend/web/components/release-pipelines/CreateReleasePipeline.tsx

```diff
@@ -158,27 +162,43 @@ function CreateReleasePipeline() {
     return pipelineData.stages.every(validateStage)
   }
 
-  const handleSave = () => {
-    createReleasePipeline({
+  const handleSubmit = async (id?: number) => {
+    if (id) {
+      return await updateReleasePipeline({
+        id,
+        name: pipelineData.name,
+        project: Number(projectId),
+        stages: pipelineData.stages.map((stage, index) => ({
+          ...stage,
+          order: index,
+        })),
+      }).unwrap()
+    }
+
+    return await createReleasePipeline({
       name: pipelineData.name,
       project: Number(projectId),
       stages: pipelineData.stages.map((stage, index) => ({
         ...stage,
         order: index,
       })),
-    })
+    }).unwrap()
   }
 
-  const handleUpdate = (id: number) => {
-    updateReleasePipeline({
-      id,
-      name: pipelineData.name,
-      project: Number(projectId),
-      stages: pipelineData.stages.map((stage, index) => ({
-        ...stage,
-        order: index,
-      })),
-    })
+  const handleSubmitAndPublish = async (id?: number) => {
+    try {
+      const response = await handleSubmit(id)
+      await publishReleasePipeline({
+        pipelineId: response.id,
+        projectId: Number(projectId),
+      })
+      toast(`Release pipeline published successfully`)
```

Comment:
It triggers to popup 
<img width="408" height="229" alt="image" src="https://github.com/user-attachments/assets/11e3cc92-eb63-419e-912f-73d68f411b36" />

Maybe we should just silence the first one when publishing straight ?

Reply:
updated